tsconfig.json

```
@@ -1,110 +1,44 @@
-        // "strictPropertyInitialization": true,             /* Check for class properties that are declared but not set in the constructor. */
-        // "strictBuiltinIteratorReturn": true,              /* Built-in iterators are instantiated with a 'TReturn' type of 'undefined' instead of 'any'. */
-        // "noImplicitThis": true,                           /* Enable error reporting when 'this' is given the type 'any'. */
-        // "useUnknownInCatchVariables": true,               /* Default catch clause variables as 'unknown' instead of 'any'. */
-        // "alwaysStrict": true,                             /* Ensure 'use strict' is always emitted. */
-        // "noUnusedLocals": true,                           /* Enable error reporting when local variables aren't read. */
-        // "noUnusedParameters": true,                       /* Raise an error when a function parameter isn't read. */
-        // "exactOptionalPropertyTypes": true,               /* Interpret optional property types as written, rather than adding 'undefined'. */
-        // "noImplicitReturns": true,                        /* Enable error reporting for codepaths that do not explicitly return in a function. */
-        // "noFallthroughCasesInSwitch": true,               /* Enable error reporting for fallthrough cases in switch statements. */
-        // "noUncheckedIndexedAccess": true,                 /* Add 'undefined' to a type when accessed using an index. */
-        // "noImplicitOverride": true,                       /* Ensure overriding members in derived classes are marked with an override modifier. */
-        // "noPropertyAccessFromIndexSignature": true,       /* Enforces using indexed accessors for keys declared using an indexed type. */
-        // "allowUnusedLabels": true,                        /* Disable error reporting for unused labels. */
-        // "allowUnreachableCode": true,                     /* Disable error reporting for unreachable code. */
-
-        /* Completeness */
-        // "skipDefaultLibCheck": true,                      /* Skip type checking .d.ts files that are included with TypeScript. */
-        "skipLibCheck": true /* Skip type checking all .d.ts files. */
+        // File Layout
+        // "rootDir": "./src",
+        // "outDir": "./dist",
+        "tsBuildInfoFile": "./node_modules/.cache/.tsbuildinfo",
+
+        // Environment Settings
+        // See also https://aka.ms/tsconfig/module
+        "module": "preserve",
+        "moduleResolution": "bundler",
+        "target": "esnext",
+        "resolveJsonModule": true,
+
+        // Other Outputs
+        // "sourceMap": true,
+        // "declaration": true,
+        // "declarationMap": true,
+        "noEmit": true,
+
+        // Stricter Typechecking Options
+        // "noUncheckedIndexedAccess": true, // TODO: Re-enable this later
+        // "exactOptionalPropertyTypes": true,
```

Rules which are commented out should show the default value:
```suggestion
        // "exactOptionalPropertyTypes": false,
```